paparazzi/paparazzi-gradle-plugin/src/test/projects/compose/src/test/java/app/cash/paparazzi/plugin/test/ComposeReferenceLeakTest.kt

```diff
@@ -0,0 +1,40 @@
+package app.cash.paparazzi.plugin.test
+
+import androidx.compose.ui.platform.ComposeView
+import app.cash.paparazzi.Paparazzi
+import org.junit.AfterClass
+import org.junit.Rule
+import org.junit.Test
+import java.lang.ref.WeakReference
+
+class ComposeReferenceLeakTest {
+  @get:Rule
+  val paparazzi = Paparazzi()
+
+  @Test
+  fun cleansUpComposeReferences() {
+    weakComposeView = WeakReference(
+      ComposeView(paparazzi.context).apply {
+        setContent {
+          HelloPaparazzi()
+        }
+
+        paparazzi.snapshot(this)
+      }
+    )
+  }
+
+  companion object {
+    private lateinit var weakComposeView: WeakReference<ComposeView>
+
+    @AfterClass
+    @JvmStatic
+    fun teardown() {
+      assert(weakComposeView.get() != null)
+
+      System.gc()
+
+      assert(weakComposeView.get() == null)
+    }
```

Comment:
@gamepro65 This is flaky, it could be freed up any time depending on when GC executes. We can't assert that it's non-null because it's a weak reference!

I got a random failure: https://github.com/TWiStErRob/paparazzi/actions/runs/3967102579/jobs/6798680398
```
java.lang.AssertionError: Assertion failed
	at app.cash.paparazzi.plugin.test.ComposeReferenceLeakTest$Companion.teardown(ComposeReferenceLeakTest.kt:33)
	at app.cash.paparazzi.plugin.test.ComposeReferenceLeakTest.teardown(ComposeReferenceLeakTest.kt)
```

Reply:
It is possible to keep this assertion but then it's necessary to fully control the references:
```kotlin
class ComposeReferenceLeakTest {
  @get:Rule
  val paparazzi = Paparazzi()

  @Test
  fun cleansUpComposeReferences() {
    composeView = ComposeView(paparazzi.context).apply {
      setContent {
        HelloPaparazzi()
      }
      paparazzi.snapshot(this)
    }
    weakComposeView = WeakReference(composeView)
  }

  companion object {
    private var composeView: ComposeView? = null
    private lateinit var weakComposeView: WeakReference<ComposeView>

    @AfterClass
    @JvmStatic
    fun teardown() {
      assert(weakComposeView.get() != null)

      composeView = null
      System.gc()

      assert(weakComposeView.get() == null)
    }
```
this would still fail if there's another reference held somewhere.

Reply:
😅  Will update the test, thanks for calling out!

Reply:
Nice response, thanks for fixing.